### ISSUE TITLE
ToString has invalid/no console output after any property, method or function is accessed

### ISSUE THREAD

**Issue by agowa** [PowerShell/PowerShell#20694]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Create a powershell class with an override for ToString(). Call any property, method or function on it. And finally output it to the console again. See below for an example script including its console output as comments.

### Expected behavior

```console
It always calling the ToString() and always correctly forming console output regardless of what function, method, or property was accessed previously.
```

### Actual behavior

```console
class TestClass {
  hidden [bool]$_foo = $false

  [void]Enable() {
    $this._foo = $true
  }

  [void]Disable() {
    $this._foo  = $false
  }

  [string]ToString() {
    return ('Value is: {0}' -f $this._foo.ToString())
  }
}

#1
[Collections.Generic.Dictionary[int, TestClass]]$testClassDict = @{}
1..4 | ForEach-Object { $testClassDict[$_] = [TestClass]::new() }
$testClassDict
<#
Key Value
--- -----
  1 Value is: False
  2 Value is: False
  3 Value is: False
  4 Value is: False
#>
$testClassDict[1]
# Value is: False
$testClassDict[1].ToString()
# Value is: False
$testClassDict[1]
#
#
$testClassDict
<#
Key Value
--- -----
  1 Value is: False
  2 Value is: False
  3 Value is: False
  4 Value is: False
#>



#2
[TestClass]$_testClass = [TestClass]::new()
$_testClass
# Value is: False
$_testClass | Write-Host
# Value is: False
$_testClass.Enable()
$_testClass | Write-Host
# Value is: False
$_testClass
#
#
$_testClass.ToString()
# Value is: False
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.9
PSEdition                      Core
GitCommitId                    7.3.9
OS                             Microsoft Windows 10.0.17763
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

![image](https://github.com/PowerShell/PowerShell/assets/2544867/6f81b7fd-29cf-4430-b811-455f5f320728)

**Comment by mklement0** [comment]
The bug appears to be related to `PSReadLine` in some fashion, but I'm unclear on how:

Using the following minimal repro, I see it _only_ :
 * with custom `class` definitions that have _no (non-hidden) properties_,
 * if the following commands are submitted _interactively_, _one by one_ (which _pasting_ the block of lines does)
 * _and_ if `PSReadLine` is loaded at the time:

```powershell
# Paste this block of lines *at the prompt*:
# You should see TWO "Foo" output lines, but the second implicit output of $o 
# prints just an empty line - as a result of the attempted property access with $o.no
class Foo {  }; $o = [Foo]::new()
$o
$o.no
$o
```
* It is the (attempted) member access `$o.no` that triggers the problem (it doesn't matter whether the member exists).
* If I execute `Remove-Module PSReadLine` *first*, the problem does _not_ surface.
* If do it _after_, the symptom persists with the `$o` object; conversely, an unaffected `$o` instance created while `PSReadLine` was _not_ loaded continues to be unaffected after reloading `PSReadLine`.
* I _never_ see the problem if the code is executed via a _script file_ (whether with dot-sourcing or not).
* On a general note:
  * It is only because your `class` has _no (non-hidden) properties_ that `.ToString()` is even called by the formatting system.
  * Normally, object's _properties_ are enumerated and displayed, as happens with `[pscustomobject] @{ foo = 1 }`, for instance; curiously, `[pscustomobject] @{  }` produces only an _empty line_, i.e. the same symptom as with the bug here, thought it appears to the unrelated bug of direct calls to `.ToString()` on `[pscustomobject]` instances returning the _empty string_ - see #6163

**Comment by B-Art** [comment]
Looks like your are running it in vscode. That is a different ballgame.
Is the behavior the same in Pwsh itself?

**Comment by mklement0** [comment]
No, I was running it in regular console windows, and I see it everywhere, including in Visual Studio Code, and in Terminal.app and iTerm2.app on macOS.

However:
* My instructions were flawed: I neglected to mention that _on Windows_ you have to paste _via right-clicking_, as only then are the commands submitted _one by one_.
* More importantly: The problem seems **connected to my specific PSReadLine configuration** - I have yet to discover what, specifically, triggers the problem. Having PSReadLine loaded _alone_ does _not_ trigger the problem.

I've update my previous comment.

**Comment by mklement0** [comment]
@B-Art, I missed that you're not the OP (and neither am I).
* Yes, the screenshot implies use of Visual Studio Code's integrated terminal.
* However, I see no difference in behavior depending on the console/terminal environment.
* I do see that it _works by default_, but - for reasons still unknown to me - _doesn't work with my custom PSReadLine configuration_, regardless of environment.

**Comment by B-Art** [comment]
Than maybe (to investigate further) you must share the way you changed your psreadline settings?

Otherwise it will be hard to reproduce.

**Comment by mklement0** [comment]
@B-Art, the pasting part is incidental to the problem: what matters is whether the commands are submitted _one by one_.

I've narrowed down the problem, and it is only indirectly related to `PSReadLine`: it is its `CompletionPredictor` plugin module whose import causes the problem. I've updated my original comment.
